lib/iris/fileformats/nimrod_load_rules.py

```diff
@@ -317,8 +700,175 @@ def add_attr(name):
+
+    source = field.source.strip()
+    rematcher = re.compile(r"^ek\d\d$")
+    if rematcher.match(source) is not None or source.find("umek") == 0:
+        source = "MOGREPS-UK"
+    cube.attributes["source"] = source
+    cube.attributes["title"] = "Unknown"
+    cube.attributes["institution"] = "Met Office"
+
+
+def known_threshold_coord(field):
+    """
+    Supplies known threshold coord meta-data for known use cases.
+    """
+    coord_keys = {}
+    if field.field_code == 161 and field.threshold_value >= 0.0:
+        coord_keys = {"var_name": "threshold"}
+        if field.threshold_value_alt > 8.0:
+            coord_keys["standard_name"] = "height"
+            coord_keys["units"] = "metres"
+        else:
+            coord_keys["standard_name"] = "cloud_area_fraction"
+            coord_keys["units"] = "oktas"
+    if field.field_code == 29 and field.threshold_value >= 0.0:
+        if is_missing(field, field.threshold_type):
+            coord_keys = {
+                "standard_name": "visibility_in_air",
+                "var_name": "threshold",
+                "units": "metres",
+            }
+        else:
+            coord_keys = {
+                "long_name": "fog_fraction",
+                "var_name": "threshold",
+                "units": "1",
+            }
+    if (
+        field.field_code == 422
+        and field.threshold_value >= 0.0
+        and is_missing(field, field.threshold_type)
+    ):
+        coord_keys = {"long_name": "radius_of_max", "units": "km"}
+    if field.field_code == 821:
+        coord_keys = {
+            "standard_name": "wind_speed_of_gust",
+            "var_name": "threshold",
+            "units": "m/s",
+        }
+    return coord_keys
 
-    cube.attributes["source"] = field.source.strip()
+
+def probability_coord(cube, field):
+    """
+    Adds a coord relating to probability meta-data from the header to the
+    cube if appropriate.
+    Returns True if this is a blended multi-member field
+    """
+    probtype_lookup = {
+        1: {
+            "var_name": "threshold",
+            "attributes": {"relative_to_threshold": "above"},
+        },
+        2: {
+            "var_name": "threshold",
+            "attributes": {"relative_to_threshold": "below"},
+        },
+        3: {"long_name": "percentile", "units": "1"},
+        4: {
+            "var_name": "threshold",
+            "attributes": {"relative_to_threshold": "equal"},
+        },
+    }
+    probmethod_lookup = {
+        1: "AOT (Any One Time)",
+        2: "ST (Some Time)",
+        4: "AT (All Time)",
+        8: "AOL (Any One Location)",
+        16: "SW (Some Where)",
+    }
+    is_multi_member_field = False
+    coord_keys = probtype_lookup.get(field.threshold_type, {})
+    if coord_keys:
+        is_multi_member_field = True
+    coord_keys.update(known_threshold_coord(field))
+    if not coord_keys.get("units", None):
+        coord_keys["units"] = DEFAULT_UNITS.get(field.field_code, None)
```

Comment:
Are you confident this acts as wanted?
The 'units' keyword defaults to '1' for coords, not None.
Owing to the odd way units are validated (via cf_units), if 'field.field_code' is not recognised in DEFAULT_UNITS, this code will result in units="unknown" rather that the usual default units='1' (as if 'units' were not passed).
If that's not what you wanted, it would be safer to omit 'units' from coord_keys here, and allow the Coord constructor to assign a default.

Reply:
Note: likely to be impacted by future changes, [as indicated below](https://github.com/SciTools/iris/pull/3647#issuecomment-629232629)

Reply:
I've changed this to specify "unknown" as "1" is not reasonable. I've also added a warning when this gets triggered.